### PR TITLE
fix array builder alert issues

### PR DESCRIPTION
**Pull request by rhasselle-oddball**
## Summary

- Fix alert issues for array builder
- Fix focus

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team-forms#1281

## Testing done

- Browser testing

## Screenshots

image

## What areas of the site does it impact?

Array builder alerts

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/platform/forms-system/src/js/patterns/array-builder/ArrayBuilderSummaryPage.jsx [hunk 1]
```diff
@@ -16,21 +16,19 @@ import {
...
-        status="success"
-        closeBtnAriaLabel="Close notification"
-        uswds
-      >
-        {text}
-      </VaAlert>
-    </div>
-  ),
+const SuccessAlert = ({ nounSingular, index, onDismiss, text }) => (
+  <div className="vads-u-margin-top--2">
```
Comment (Thrillberg) [inline src/platform/forms-system/src/js/patterns/array-builder/ArrayBuilderSummaryPage.jsx:20]: Always glad to have `ref`s removed!!
EDIT: Ah, I see it was just moved. Well, still, seems better how you have it here 😄

File: src/platform/forms-system/src/js/patterns/array-builder/ArrayBuilderSummaryPage.jsx [hunk 2]
```diff
@@ -126,18 +124,24 @@ export default function ArrayBuilderSummaryPage({
...
-            timeout = setTimeout(() => {
-              scrollAndFocus(updatedAlertRef.current);
-            }, 300);
-          }
+        }
+      },
+      [updatedNounSingular, updateItemIndex],
+    );
+
+    useEffect(
```
Comment (Thrillberg) [inline src/platform/forms-system/src/js/patterns/array-builder/ArrayBuilderSummaryPage.jsx:134]: Nice refactor

File: src/platform/forms-system/src/js/patterns/array-builder/ArrayBuilderSummaryPage.jsx [hunk 3]
```diff
@@ -275,7 +300,11 @@ export default function ArrayBuilderSummaryPage({
           ) : (
             <>
               <div className="form-review-panel-page-header-row">
-                <h4 className="form-review-panel-page-header vads-u-font-size--h5">
+                <h4
+                  className="form-review-panel-page-header vads-u-font-size--h5"
+                  data-title-for-noun-singular={`${nounSingular}`}
+                  f
```
Comment (Thrillberg) [inline src/platform/forms-system/src/js/patterns/array-builder/ArrayBuilderSummaryPage.jsx:306]: What's up with this `f`?

**Review by Thrillberg — DISMISSED** [review]
Looks great! Nice work!

File: src/platform/forms-system/src/js/patterns/array-builder/ArrayBuilderSummaryPage.jsx (hunk 3 above)
Comment (rhasselle-oddball) [reply src/platform/forms-system/src/js/patterns/array-builder/ArrayBuilderSummaryPage.jsx]: oops. fixed!

**Review by Thrillberg — APPROVED** [review]

**Review by powellkerry — APPROVED** [review]

**Pull request merged**